### DISCUSSION TITLE
Invalid now.json - `files` Invalid request: `files` should NOT have more than 10000 items..

### DISCUSSION THREAD

**Discussion by bigsee** [vercel/vercel#5464] [Help]
I have a small Nuxt site deployed on Vercel which has been there for ages. I haven't deployed it for probably six months.

Just tried to make a simple text change and re-deploy and now I get the following error. The old now.json was something like this:

``` js
{
  "name": "my-project",
  "version": 2,
  "alias": [
    "my-project.co.uk",
    "www.my-project.co.uk"
  ],
  "builds": [
    {
      "src": "nuxt.config.js",
      "use": "@nuxtjs/now-builder",
      "config": {
        "serverFiles": [
          "middleware/**",
          "plugins/**",
          "lib/**"
        ]
      }
    }
  ],
  "routes": [
    { "src": "/_nuxt/.+", "headers": { "cache-control": "s-maxage=31536000" } },
    { "src": "^/(.*)", "dest": "/" }
  ],
  "public": false
}
```

I've since tried replacing `@nuxtjs/now-builder` with `@nuxtjs/vercel-builder`. I've also tried zero config with no `vercel.json` or `now.json` and every combination in between. I make sure to delete all nuxt and vercel folders and caches between each attempt. Every time I get the same error:

`Invalid now.json - 'files' Invalid request: 'files' should NOT have more than 10000 items..`

 There is also a link to the docs here: [https://vercel.com/docs/configuration#project/files](https://vercel.com/docs/configuration#project/files) but this doesn't yield anything obvious.

What am I doing wrong?

**Comment by bigsee** [comment]
\*bump\*

Just wondering if anyone has any feedback on this? Feels like a bug to me and the docs link in the area contain no information on the subject that I can see.

**Comment by bigsee** [comment]
I have not been able to get this working so I've now submitted a formal support request. If anyone else has any ideas in the meantime, I'd be grateful to read them.

**Reply by cobrexus** [reply]
I just noticed that the link you were provided with doesn't actually go anywhere. There is no `files` heading on that page in the docs. This is worth pointing out to the support team.

**Reply by bigsee** [reply]
I will do that

**Comment by cobrexus** [comment]
Some of the fields you are using are now deprecated:
* [`name`](https://vercel.com/docs/configuration#project/name): use [Project Linking](https://vercel.com/docs/cli#commands/overview/project-linking) instead.
* [`version`](https://vercel.com/docs/configuration#project/version): since ZEIT 1.0 is shut down, this always defaults to `2` so should not be set anyway,
* [`alias`](https://vercel.com/docs/configuration#project/alias): to assign a custom Domain to your Project, [define it in the Project Settings](https://vercel.com/docs/custom-domains) instead.
* [`routes`](https://vercel.com/docs/configuration#project/routes): to customize the platform's routing behaviour, use [`cleanUrls`](https://vercel.com/docs/configuration#project/cleanurls), [`trailingSlash`](https://vercel.com/docs/configuration#project/trailingslash), [redirects](https://vercel.com/docs/configuration#project/redirects), [rewrites](https://vercel.com/docs/configuration#project/rewrites), and/or [headers](https://vercel.com/docs/configuration#project/headers) instead.
* `public` is `false` by default, so no need to specify it (it is not deprecated, however)

Also, the `now.json` file should be renamed to `vercel.json`, because it may be deprecated soon.

Make sure you have the latest version of the Vercel CLI installed: `vercel -v` should return `21.0.1`. If not, update with `npm i -g vercel`.

**Reply by bigsee** [reply]
@AryanBeezadhur thank you for your helpful response. I am aware that some of the fields are deprecated and need updating soon: `name` and `version` should not affect the deployment or result in the behaviour I'm seeing though, unless I'm missing something?

With regard to `alias`, to my knowledge, while it is recommended to use the project settings, my use case meant I needed custom domains for different environments (e.g. staging.mysite.com). And aliases were the only obvious way to achieve this without the Github integration, which had a bug that I reported, which broke my deployments until I removed it. 

`routes` is another case entirely. I see that this is no longer the _recommended_ approach for the use case I had. However, I also see that it is not deprecated. Therefore, I don't think it unreasonable to expect deployment to work as before especially since this is a simple text change.

**Reply by bigsee** [reply]
P.S. As mentioned, I also tried zero config but the same error is returned.

**Comment by paulogdm** [comment]
@bigsee Is it possible that you are deploying unwanted files, like `node_modules`?

**Reply by bigsee** [reply]
@paulogdm hmmm... `node_modules` has always been in my `.gitignore`. I will double-check the source...

**Reply by bigsee** [reply]
@paulogdm double-checked this: `node_modules` is definitely in `.gitignore`. I also noticed that `vercel dev` now seems to create a `node_modules_dev` folder so I added that too, with the same result.

For a while `vercel dev` would not work. After deleting `node_modules`, doing a fresh `npm install` and then running `vercel dev`, I would get the following error:

```
Using npm                                                                                                                                 17:06:48
ℹ Prepare build took: 2.790129 ms                                                                                                         17:06:48
 ----------------- Install devDependencies -----------------                                                                              17:06:48
Error! EPERM: operation not permitted, unlink 'node_modules'
```

When I delete `node_modules` and run `vercel dev` _without_ installing beforehand, the command correctly serves the site locally.

But then running `vercel` gives the same result. I have a support ticket open so I will provide repo access there and post a solution here when we find it.

**Comment by giancarlosisasi** [comment]
I'm having the same error using the vercel cli 😢 . 
is there a solution or explanation for this error? 😢

**Comment by lonyele** [comment]
On my local machine using vercel-cli it works fine, but on self-managed gitlab's CI, it fails. 
local env. `node: v16.13.0`, `mac os: Monterey 12.2.1`,`pnpm@6.30.0`
self-managed gitlab ci env. `node:lts-alpine`, `ubuntu 20.04`, `pnpm@6.30.0`

<img width="1410" alt="스크린샷 2022-04-18 오후 7 55 01" src="https://user-images.githubusercontent.com/15892571/163798957-ddf6e86c-6a05-4d8c-8d2a-b35197f67d2e.png">

**Reply by lonyele** [reply]
Ok... I succeeded from self-managed gitlab ci. 
My guess is that the problem is the npm/pnpm cache(or it could be node_modules not sure). 
Ususally npm/yarn/pnpm installed output will be put into some cache store(in my case pnpm-store) and later can be used on test, lint, deploy or whatever stage/pipeline afterward. I just tried to track down the issue and remove all pnpm cache things and it worked. I need to add cache feature any since it is very needed but yeah, it was the problem for me. Hope others can get help

**Reply by theHarisKhan** [reply]
hey! i'm facing the same issue my files say it exceed 18256,
Let me know how you resolved this issue,

**Comment by praisegeek** [comment]
I experienced this issue despite having .gitignore exclude node_modules from source. But when I added .vercelignore to my project root directory with the contents
```
node_modules 
.cache
.output
/build/
/public/build
/api/index.js
```
the error left. I believe if I had added /build and /public/build to .gitignore, it would have worked still.

**Reply by Daavidaviid** [reply]
Great! I'm using react-native-web, I needed to add all the folders like .expo, ios, android, etc

**Reply by TheBlueFriend** [reply]
So .vercelignore does the work! It was working fine until one day when the error showed up. They should debug clearing the cache properly.

**Comment by zamgill** [comment]
This fix worked for me:

> Hi @dpcdev1999, can you try `vercel deploy --prebuilt --archive=tgz` and let us know how it goes?

https://github.com/vercel/vercel/issues/8571#issuecomment-1259816713
